dashboard/app/models/sections/omni_auth_section.rb

```diff
@@ -27,18 +27,33 @@
 
 class OmniAuthSection < Section
   def self.from_omniauth(code:, type:, owner_id:, students:)
-    oauth_section = where(code: code).first_or_create! do |section|
+    oauth_section = with_deleted.where(code: code).first_or_create! do |section|
       section.name = 'New Section'
       section.user_id = owner_id
       section.login_type = type
     end
-    students.each do |student|
-      oauth_section.add_student User.from_omniauth(student, {'user_type' => User::TYPE_STUDENT})
+
+    oauth_section.restore if oauth_section.deleted?
```

Comment:
Deleting (e.g., soft-deleting) a section causes all followers to be (soft) deleted. Doing `restore` recovers only the section. Doing `restore(recursive: true, recovery_window: 5.minutes)` causes all followers deleted within five minutes of the section delete to also be restored.